src/core/eme/handle_session_events.ts

```diff
@@ -141,6 +149,14 @@ export default function handleSessionEvents(
 
       // find out possible errors associated with this event
       (session.keyStatuses as any).forEach((keyStatus : string, keyId : string) => {
+        if (keyStatus === KEY_STATUS_EXPIRED || keyId === KEY_STATUS_EXPIRED) {
+          const error =
+            new EncryptedMediaError("KEY_STATUS_CHANGE_ERROR", "expired", false);
+          sessionWarningSubject$.next({
+            type: "warning",
+            value: error,
+          });
+        }
```

Comment:
Why use a subject here? The point of relying on true observables only is that it's immediately clear that the flow will stop if a user stops on a warning.

Also, aren't we breaking the API here? We might want to do this only if a new option is set. Throw otherwise